src/structures/Guild.js

```diff
@@ -101,6 +101,14 @@ class Guild extends Base {
      * @type {number}
      */
     this.shardID = data.shardID;
+
+    if ('nsfw' in data || typeof this.nsfw !== 'boolean') {
+      /**
+       * Whether the guild is designated as not safe for work
+       * @type {boolean}
+       */
+      this.nsfw = Boolean(data.nsfw);
+    }
```

Comment:
Seems like this logic is from User#bot where Discord will only send the property when it's `true`? Discord always sends `nsfw` for full guild objects.

```suggestion
    if ('nsfw' in data) {
      /**
       * Whether the guild is designated as not safe for work
       * @type {boolean}
       */
      this.nsfw = data.nsfw;
    }
```

Reply:
do they? we've had that conversation before and essentially all booleans can act like that

Reply:
well it's documented as required and it's present when it's false; afaik this handling is only needed when it's documented as optional